syncall/tw_caldav_utils.py

```diff
@@ -0,0 +1,177 @@
+        key
+        for key, value in aliases_tw_caldav_status.items()
+        if caldav_item["status"] == value
+    )
+
+    # Priority
+    if caldav_item.get("priority"):
+        priority = next(
+            (
+                key
+                for key, value in aliases_tw_caldav_priority.items()
+                if caldav_item["priority"] == value
+            ),
+            None,
+        )
+        if priority:
+            tw_item["priority"] = priority
+
+    # Timestamps
+    if "last-modified" in caldav_item.keys():
+        tw_item["modified"] = caldav_item["last-modified"]
+
+    # Start/due dates
+    if "due" in caldav_item.keys():
+        tw_item["due"] = caldav_item["due"]
+
+    if "categories" in caldav_item.keys():
+        tw_item["tags"] = caldav_item["categories"]
+
+    return tw_item
+
+
+def map_ics_to_item(vtodo) -> Dict:
+    """
+    Utility class that will extract the relevant info from an icalendar_component into a python dict
+    """
+    todo_item = {}
+    todo_item["id"] = str(vtodo.get("uid"))
+    for i in ["status", "priority", "description", "summary"]:
+        item = vtodo.get(i)
+        if item:
+            todo_item[i] = str(item).lower()
+    for i in ["last-modified"]:
+        item = vtodo.get(i)
+        if item:
+            todo_item[i] = item.dt
+
+    if vtodo.get("due"):
+        todo_item["due"] = vtodo["due"].dt
+
+    if vtodo.get("categories"):
+        todo_item["categories"] = [str(category) for category in vtodo["categories"].cats]
+
+    return todo_item
+
+
+def _parse_caldav_item_desc(
+    caldav_item: Item,
+) -> Tuple[List[str], Optional[UUID]]:
+    """
+    Parse and return the necessary TW fields off a caldav Item.
+
+    Pretty much directly copied from tw_gcal_utils, however we handle status differently, so only return annotations/uuid
+    """
+    annotations: List[str] = []
+    uuid = None
+
+    if "description" not in caldav_item.keys():
+        return annotations, uuid
+
+    caldav_desc = caldav_item["description"]
+    # strip whitespaces, empty lines
+    lines = [line.strip() for line in caldav_desc.split("\n") if line][1:]
+
+    # annotations
+    i = 0
+    for i, line in enumerate(lines):
+        parts = line.split(":", maxsplit=1)
+        if len(parts) == 2 and parts[0].lower().startswith("* annotation"):
+            annotations.append(parts[1].strip())
+        else:
+            break
+
+    if i == len(lines):
+        return annotations, uuid
+
+    # Iterate through rest of lines, find only the uuid
+    for line in lines[i:]:
+        parts = line.split(":", maxsplit=1)
+        if len(parts) == 2 and parts[0].lower().startswith("* uuid"):
+            try:
+                uuid = UUID(parts[1].strip())
+            except ValueError as err:
+                logger.error(
+                    f'Invalid UUID "{err}" provided during caldav -> TW conversion,'
+                    f" Using None...\n\n{traceback.format_exc()}"
```

Comment:
What does it mean for the UUID to be None?

Reply:
I'm afraid this was just shamelessly copied from here: https://github.com/bergercookie/syncall/blob/master/syncall/tw_gcal_utils.py#L228

Not sure of the wider implications on a given sync, I guess it would create a new TW item if it can't find the associated UUID?